### DISCUSSION TITLE
Helper functions for basic structure organization

### DISCUSSION THREAD

**Discussion by mitchuman** [sanity-io/sanity#3389] [Sanity Studio Feedback]
I've set up [the following helper functions](https://gist.github.com/mitchuman/95842862783c3b0cea9160a8c557acef) that I use in every project to quickly organize documents to `singletons`, `lists` and `groups`.

(I've updated this now to work with v3, where `S` is passed down as an argument.)

It would be nice to have similar helper utilities built into or provided by default.

### Example implementation:

```js
const AllDocuments = S.documentTypeListItems()
  .filter(item => ![
    'site',
    'header', 'footer',
    'page',
    'blogPg', 'blogCategory', 'article',
    'clinicalStudy',
  ].includes(item.getId()))

export default () =>
  S.list()
    .title('Content')
    .items([
      group('Site settings', [
        singleton('Site', 'site').icon(BsInfoCircleFill),
        singleton('Header', 'header').icon(RiLayoutTop2Line),
        singleton('Footer', 'footer').icon(RiLayoutBottom2Line),
      ]).icon(BsInfoCircleFill),

      list('Pages', 'page'),

      group('Blog', [
        list('Blog pages', 'blogPg'),
        list('Categories', 'blogCategory'),
        list('Articles', 'article'),
      ]).icon(FaPenAlt),

      list('Clinical Studies', 'clinicalStudy'),

      ...AllDocuments,
    ])
```

**Comment by legowhales** [comment]
Hey, I see that you reverted back to not passing S down as an argument. How does it work in v3 for you?
I have similar helpers, and I was curious if I had to change them for v3 :-).

Thank's for sharing your helper functions btw!

**Reply by mitchuman** [reply]
I've reverted back since I was only trying out v3 temporarily—the previous version with `S` being passed as an argument should still work.

**Reply by JanneHelewaut** [reply]
Do you have an example of the S you are passing down. 
I keep running into issues that the structure-builder wants parts defined, and I can't figure out how I can make it work! 

Thank you!

**Reply by mitchuman** [reply]
The current version is intended for v2. For v3, take a look at [the previous revision here](https://gist.github.com/mitchuman/b5d56d648eeb55cc9974298a8d294129/revisions#) where it's basically being passed from the parent (in my `structure.js` or the desk structure config file).

Note: make sure to remove the import statement for `S`.
